src/index.js

```diff
@@ -2,7 +2,7 @@ const server = require('./server');
 const puppeteer = require('./puppeteer');
 const defaults = require('../config/defaults');
 
-module.exports = async function getRunner(siteKey, constructorOptions = defaults) {
+module.exports = async function getRunner(siteKey, userName, constructorOptions = defaults) {
```

Comment:
This would break the current API, I would rather send the `userName` in the `constructorOptions`, and set a default `userName: null` in the `defaults` object, ie:

```js
module.exports = async function getRunner(siteKey, constructorOptions = defaults) {
...
puppeteer({
  siteKey,
  userName: options.userName,
  interval: options.interval,
  port: options.port,
  host: options.host
...
```

Otherwise the user has to send `null` as the second parameter if they want to use `constructorOptions`...

What do you think?

Reply:
👍 use contructorOptions it's meant to add options. See https://github.com/cazala/coin-hive/pull/10/files

Reply:
Hey @cazala I am not sure if it would break anything. I tested it seemed to work (but don't take my word for it - its your project, so maybe better to go with the constructorOptions).

I am abit confused how those constructorOptions work, how do you sent them exactly? (I checked the Readme but wasn't very clear).